plugin-server/src/utils/fetch.ts

```diff
@@ -88,16 +26,97 @@ export async function raiseIfUserProvidedUrlUnsafe(url: string): Promise<void> {
     if (parsedUrl.protocol !== 'http:' && parsedUrl.protocol !== 'https:') {
         throw new FetchError('Scheme must be either HTTP or HTTPS', 'posthog-host-guard')
     }
+    return parsedUrl
+}
+
+function isGlobalIPv4(ip: ipaddr.IPv4): boolean {
+    const octets = ip.octets
+    return !(
+        (
+            octets[0] === 0 || // "This network" (0.0.0.0/8)
+            ip.range() !== 'unicast' || // Non-unicast addresses
+            octets[0] === 127 || // Loopback (127.0.0.0/8)
+            (octets[0] === 169 && octets[1] === 254) || // Link-local (169.254.0.0/16)
+            (octets[0] === 255 && octets[1] === 255 && octets[2] === 255 && octets[3] === 255)
+        ) // Broadcast
+    )
+}
```

Comment:
super nit: I find that somewhat hard to read with the large condition negation but rather personal preference. I'd much rather go with 

```suggestion
function isGlobalIPv4(ip: ipaddr.IPv4): boolean {
    const [a, b, c, d] = ip.octets

    if (a === 0) return false // "This network" (0.0.0.0/8)
    if (ip.range() !== 'unicast') return false // Non-unicast addresses
    if (a === 127) return false // Loopback (127.0.0.0/8)
    if (a === 169 && b === 254) return false // Link-local (169.254.0.0/16)
    if (a === 255 && b === 255 && c === 255 && d === 255) return false // Broadcast

    return true
}

no need to change tho


```

Reply:
Wasn't actually as bad once i moved the comments around

Reply:
thanks!